BJS2-77591 & BJS2-77587 | Создание платежного счета & Работа с балансами

Anatolii-Izotov opened the pull request:
Описание задачи [BJS2-77591](https://faang-school.atlassian.net/jira/software/c/projects/BJS2/boards/487/backlog?selectedIssue=BJS2-77591) и сущностей, которые были добавлены: [ссылка](https://docs.google.com/document/d/1fzTb9plVfQePhlKfnwlkUNshmEEZGpj4cY9_WyaHUzE/edit?tab=t.0)

[BJS2-77591]: https://faang-school.atlassian.net/browse/BJS2-77591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Anatolii-Izotov committed:
feat(payment-account-creating): Add `Account` and `AccountBalance` models with related enums and database migration scripts. Include initial data seeding for accounts and balances [BJS2-77591]

Anatolii-Izotov committed:
feat: CI

AlexeyAntipin commented on build.gradle.kts:
```diff
@@ -6,7 +6,7 @@ plugins {
 group = "faang.school"
 version = "1.0"
-java.sourceCompatibility = JavaVersion.VERSION_17
+//java.sourceCompatibility = JavaVersion.VERSION_17
```
Почему закомментировал?

AlexeyAntipin commented on src/main/resources/application.yaml:
```diff
@@ -30,13 +30,14 @@ spring:
 
   liquibase:
     change-log: classpath:db/changelog/db.changelog-master.yaml
+#    clear-checksums: true
```
Зачем?

AlexeyAntipin commented on src/main/resources/db/changelog/changeset/V001__account-service_account.sql:
```diff
@@ -1 +1,26 @@
+    owner_id bigint NOT NULL,
+    account_type varchar(32) NOT NULL,
+    status varchar(16) NOT NULL DEFAULT 'ACTIVE',
+    created_at timestamptz DEFAULT current_timestamp,
```
На мой взгляд, лучше обычные таймстэмпы использовать для хранения в базе

Anatolii-Izotov committed:
chore: Update database timestamp types, enable Java 17 compatibility, and clean up unused configurations

Anatolii-Izotov committed:
chore: Changed column type for `currency` to character(3) in table 'balances'

Anatolii-Izotov committed:
authorized balance added

Anatolii-Izotov committed:
feat(account): Implement account-features: creation, freezing, unfreezing, and closing accounts. Also implemented redis cache to get a list of accounts by Owner faster [BJS2-77591]

Anatolii-Izotov committed:
Merge branch 'features/BJS2-77591--BJS2-77587' into features/BJS2-77591-payment-account-creating

Anatolii-Izotov committed:
refactor: Move AccountCreatedEvent to event package and update imports

danbg12 committed:
Ready

danbg12 committed:
Ready

Anatolii-Izotov committed:
feat(cache): Add cache eviction mechanisms and update DTOs with constructors
- Added tests for `AccountServiceImpl` to ensure coverage of new features.

Anatolii-Izotov committed:
feat(account): Add validation for duplicate accounts, tests for AccountController, and new database indexes

- Added validation in `AccountServiceImpl` to prevent duplicate account creation.
- Introduced `AccountControllerTest` to cover controller-level scenarios.
- Updated `application-test.yaml` to disable Consul for tests.
- Added database indexes for `owner_id+owner_type` and `account_number`.

AlexeyAntipin commented:
> Только не мержи пж :)

А почему?)

Anatolii-Izotov commented:
> > Только не мержи пж :)
> 
> А почему?)

Сори, уже можно давно писал

AlexeyAntipin commented:
> Сори, уже можно давно писал

Капец ты оперативно отвечаешь)
А тут еще и задача Дэна есть? Почему тогда 2 пра?

AlexeyAntipin commented:
> > Сори, уже можно давно писал
> 
> Капец ты оперативно отвечаешь) А тут еще и задача Дэна есть? Почему тогда 2 пра?

Я про вот эту задачу https://faang-school.atlassian.net/browse/BJS2-77587?atlOrigin=eyJpIjoiN2I0YzJjMmYzMDM3NDMwOWIzZmMyODZlNDQ1OGI3ZDEiLCJwIjoiaiJ9

Anatolii-Izotov commented:
> > > Сори, уже можно давно писал
> > 
> > 
> > Капец ты оперативно отвечаешь) А тут еще и задача Дэна есть? Почему тогда 2 пра?
> 
> Я про вот эту задачу https://faang-school.atlassian.net/browse/BJS2-77587?atlOrigin=eyJpIjoiN2I0YzJjMmYzMDM3NDMwOWIzZmMyODZlNDQ1OGI3ZDEiLCJwIjoiaiJ9

Мы с ним так договаривались, но он передумал, я думаю он уже сюда ничего мержить не будет.

AlexeyAntipin commented:
> > > > Сори, уже можно давно писал
> > > 
> > > 
> > > Капец ты оперативно отвечаешь) А тут еще и задача Дэна есть? Почему тогда 2 пра?
> > 
> > 
> > Я про вот эту задачу https://faang-school.atlassian.net/browse/BJS2-77587?atlOrigin=eyJpIjoiN2I0YzJjMmYzMDM3NDMwOWIzZmMyODZlNDQ1OGI3ZDEiLCJwIjoiaiJ9
> 
> Мы с ним так договаривались, но он передумал, я думаю он уже сюда ничего мержить не будет.

Я к тому, что сейчас висит 2 пул реквеста. В одном работа с балансами (сущности, миграции и прочее). В твоей задаче все это тоже есть, но добавлена еще логика по твоей задаче. Мы же не смержим и то и то)

Anatolii-Izotov commented:
Я ему скажу

AlexeyAntipin commented:
> Я ему скажу

В общем да, договоритесь, что да как

danbg12 committed:
Merge branch 'BJS2-77587' into features/BJS2-77591--BJS2-77587

# Conflicts:
#	src/main/java/faang/school/accountservice/model/Account.java
#	src/main/java/faang/school/accountservice/model/AccountBalance.java
#	src/main/java/faang/school/accountservice/model/AccountType.java
#	src/main/resources/db/changelog/changeset/V001__account-service_account.sql
#	src/main/resources/db/changelog/changeset/V002__insert_account_and_balances.sql
#	src/main/resources/db/changelog/db.changelog-master.yaml

danbg12 committed:
Merge remote-tracking branch 'origin/features/BJS2-77591--BJS2-77587' into features/BJS2-77591--BJS2-77587

# Conflicts:
#	src/main/java/faang/school/accountservice/dto/AccountBalanceDto.java
#	src/main/java/faang/school/accountservice/mapper/AccountBalanceMapper.java
#	src/main/java/faang/school/accountservice/repository/AccountRepository.java

danbg12 committed:
Ready

danbg12 commented:
Договорились :D
